### PR TITLE
[PushNotificationIOS] Added badge icon number and alert action

### PR DESCRIPTION
**Pull request by alvaromb**
Added support for setting an optional badge icon number and alert action when using local notifications.

### CODE REVIEW

**Comment by christopherdro** [comment]
@alvaromb Any chance you can rebase this against master?
@nicklockwood Do you see anything holding this back from getting merged in?

**Comment by alvaromb** [comment]
I don't know if this got merged, completely forgot :laughing: 

I can update the PR or send another one.

**Comment by chirag04** [comment]
@alvaromb alertAction is already added in master. Do you want to rebase and add support for `applicationIconBadgeNumber`?

**Comment by alvaromb** [comment]
Ok, I'll try to do a rebase today.

**Comment by facebook-github-bot-1** [comment]
@vjeux would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

**Comment by vjeux** [comment]
I have 0 context on this but the other reviewers might

**Comment by mkonicek** [comment]
I'll try assigning this to @majak as all the iOS pull requests are assigned to Nick and this doesn't scale :)

**Comment by alvaromb** [comment]
Sorry folks, I'm currently having a gargantuan amount of work right now, didn't have time to look at this. Do you want me to rebase anything?

**Comment by mkonicek** [comment]
@alvaromb That's understandable :) Let's wait for @majak first (please reassign to someone else than Nick if you're too busy) to see how we should proceed with this PR.

**Comment by joshuapinter** [comment]
Hey Guys, I'd love to have this in here. Is there anything I can do help like a rebase against Master since alertAction is already added? Let me know.

**Comment by joshuapinter** [comment]
As a means of getting this into `master` as quick as possible (I need this for our next release) I created a new PR with the same changes on the `master` branch. 

The new PR is #6894.

Thanks for your effort @alvaromb on this! :+1:

**Comment by alvaromb** [comment]
Thanks Joshua! :)

Enviado desde mi iPhone

**Comment by mkonicek** [comment]
I'd like at least one person with Obj-C knowledge to take a look. Maybe I'll have to learn Obj-C :)

This one seems simple enough for me to review.

@facebook-github-bot shipit

**Comment by alvaromb** [comment]
Please ping me if you need some info about the PR @mkonicek 😃

**Comment by joshuapinter** [comment]
Thanks for taking a look @mkonicek. Because this PR took so long to get merged there was a bunch of code changes to master so I ended up creating a new PR with @alvaromb's changes except based off the latest master branch: #6894.

I don't know if this one or #6894 is easier for you to merge in, but they accomplish the same thing. 

Over to you! :)

File: Libraries/PushNotificationIOS/RCTPushNotificationManager.m [hunk 1]
```diff
@@ -35,6 +35,8 @@ + (UILocalNotification *)UILocalNotification:(id)json
   UILocalNotification *notification = [UILocalNotification new];
   notification.fireDate = [RCTConvert NSDate:details[@"fireDate"]] ?: [NSDate date];
   notification.alertBody = [RCTConvert NSString:details[@"alertBody"]];
+  notification.alertAction = [RCTConvert NSString:details[@"alertAction"]] ?: nil;
```
Comment (alloy) [inline Libraries/PushNotificationIOS/RCTPushNotificationManager.m]: Not a biggie, but I’m unsure what the point is of the extra ternary. Right now it reads to me like: if the result of converting is `nil` return `nil` instead.

File: Libraries/PushNotificationIOS/RCTPushNotificationManager.m [hunk 2]
```diff
@@ -35,6 +35,8 @@ + (UILocalNotification *)UILocalNotification:(id)json
...
+  notification.applicationIconBadgeNumber = [RCTConvert NSInteger:details[@"applicationIconBadgeNumber"]] ?: 0;
```
Comment (alloy) [inline Libraries/PushNotificationIOS/RCTPushNotificationManager.m]: Are you sure that you want to _clear_ the badge number if the entry was not included in the payload? That would make it effectively impossible to not touch the badge number with a payload.

**Comment by alloy** [comment]
@mkonicek I left some comments, hope it helps.

File: Libraries/PushNotificationIOS/RCTPushNotificationManager.m (hunk 2 above)
Comment (alloy) [inline Libraries/PushNotificationIOS/RCTPushNotificationManager.m]: Reading the documentation added to the JS file, that does not appear to be the intention, so this should be more like:

```objc
if (details[@"applicationIconBadgeNumber"] != nil) {
  notification.applicationIconBadgeNumber = [RCTConvert NSInteger:details[@"applicationIconBadgeNumber"]];
}
```

**Comment by alvaromb** [comment]
Thanks for the feedback @alloy. Definitely not the best code ever, I'll update the PR.

File: Libraries/PushNotificationIOS/RCTPushNotificationManager.m (hunk 2 above)
Comment (alvaromb) [inline Libraries/PushNotificationIOS/RCTPushNotificationManager.m]: Yes, this seems a better approach. I'm sending a PR in a moment.

**Comment by alvaromb** [comment]
I've added @alloy suggestions (thanks!) and updated the PR. /cc @joshuapinter

**Comment by joshuapinter** [comment]
@alvaromb Beauty! I'll go close my PR.

**Comment by alloy** [comment]
@alvaromb @mkonicek Yup, this looks good to me 👍

**Comment by mkonicek** [comment]
OK merging the updated code.

@facebook-github-bot shipit

**Comment by mkonicek** [comment]
@alvaromb Can you please rebase? There's a merge conflict with master, sorry about that.

**Comment by facebook-github-bot-3** [comment]
@alvaromb updated the pull request.

**Comment by alvaromb** [comment]
There you go @mkonicek! Please let me know if you need anything else 😃

**Comment by mkonicek** [comment]
Thanks @alvaromb!

@facebook-github-bot shipit

**Comment by facebook-github-bot-2** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/482764255217631/int_phab) to review.

**Pull request closed without merging**